simulator/tests/TestHelper.ts

```diff
@@ -0,0 +1,282 @@
+﻿import * as chai from "chai";
+import * as sinon from "sinon";
+
+import { IToken, TokenCategory } from "../../parser/interface/IToken";
+import { IParseResult } from "../../parser/interface/IParseResult";
+import { IWarrior } from "../interface/IWarrior";
+import { ITask } from "../interface/ITask";
+import { OpcodeType, ModifierType } from "../interface/IInstruction";
+import { ModeType } from "../interface/IOperand";
+import { IInstruction } from "../interface/IInstruction";
+import { ICore } from "../interface/ICore";
+import { IOptions } from "../interface/IOptions";
+
+"use strict";
+
+export default class TestHelper {
+
+    public static position = {
+        line: 1,
+        char: 1
+    };
+
+    public static buildParseResult(tokens: IToken[]): IParseResult {
+        return {
+            tokens: tokens,
+            messages: [],
+            metaData: {
+                name: "",
+                author: "",
+                strategy: ""
+            }
+
+        };
+    }
+
+    public static buildToken(category: TokenCategory, lexeme: string): IToken {
+        return {
+            category: category,
+            lexeme: lexeme,
+            position: TestHelper.position
+        };
+    }
+
+    public static instruction(
+        opcode: string,
+        modifier: string,
+        amode: string,
+        anumber: number,
+        bmode: string,
+        bnumber: number): IToken[] {
+
+        return [
+            TestHelper.buildToken(TokenCategory.Opcode, opcode),
+            TestHelper.buildToken(TokenCategory.Modifier, modifier),
+            TestHelper.buildToken(TokenCategory.Mode, amode),
+            TestHelper.buildToken(TokenCategory.Number, anumber.toString()),
+            TestHelper.buildToken(TokenCategory.Comma, ","),
+            TestHelper.buildToken(TokenCategory.Mode, bmode),
+            TestHelper.buildToken(TokenCategory.Number, bnumber.toString()),
+            TestHelper.buildToken(TokenCategory.EOL, "\n")
```

Comment:
I wonder if this kind of thing would blow up if we wrote a warrior on a different OS?